searxng/searxng
# [mod] mudularize & document searx.results

## What does this PR do?

The intention of this patch is to improve modularization & documentation of the implementations about the *result* items.

The module `searx.results` is changed into a python package with modules.

> **Note**
> This patch does not contain any functional change!

## Why is this change important?

The result types, the fields of result items and the media types of the result list are  elementary for any development of engines and templates and should therefore be documented in detail.

## How to test this PR locally?

- Compile docs `make docs.live` and jump to "Source-Code" and read "Engine Results".
- Start session `make run` and test some queries (nothing should be changed / no functional change)
